provider/cluster_rosa_classic_resource.go

```diff
@@ -0,0 +1,823 @@
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"host_prefix": {
+				Description: "Length of the prefix of the subnet assigned to each node.",
+				Type:        types.Int64Type,
+				Optional:    true,
+				Computed:    true,
+			},
+			"version": {
+				Description: "Identifier of the version of OpenShift, for example 'openshift-v4.1.0'.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"state": {
+				Description: "State of the cluster.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+		},
+	}
+	return
+}
+
+func (t *ClusterRosaClassicResourceType) NewResource(ctx context.Context,
+	p tfsdk.Provider) (result tfsdk.Resource, diags diag.Diagnostics) {
+	// Cast the provider interface to the specific implementation:
+	parent := p.(*Provider)
+
+	// Get the collection of clusters:
+	collection := parent.connection.ClustersMgmt().V1().Clusters()
+
+	// Create the resource:
+	result = &ClusterRosaClassicResource{
+		logger:     parent.logger,
+		collection: collection,
+	}
+
+	return
+}
+
+func (r *ClusterRosaClassicResource) Create(ctx context.Context,
+	request tfsdk.CreateResourceRequest, response *tfsdk.CreateResourceResponse) {
+	// Get the plan:
+	state := &ClusterRosaClassicState{}
+	diags := request.Plan.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	// Create the cluster:
+	builder := cmv1.NewCluster()
+	builder.Name(state.Name.Value)
+	builder.CloudProvider(cmv1.NewCloudProvider().ID(awsCloudProvider))
+	builder.Product(cmv1.NewProduct().ID(rosaProduct))
+	builder.Region(cmv1.NewCloudRegion().ID(state.CloudRegion.Value))
+	if !state.MultiAZ.Unknown && !state.MultiAZ.Null {
+		builder.MultiAZ(state.MultiAZ.Value)
+	}
+	if !state.Properties.Unknown && !state.Properties.Null {
+		properties := map[string]string{}
+		for k, v := range state.Properties.Elems {
+			properties[k] = v.(types.String).Value
+		}
+		builder.Properties(properties)
+	}
+
+	if !state.EtcdEncryption.Unknown && !state.EtcdEncryption.Null {
+		builder.EtcdEncryption(state.EtcdEncryption.Value)
+	}
+
+	nodes := cmv1.NewClusterNodes()
+	if !state.ComputeNodes.Unknown && !state.ComputeNodes.Null {
+		nodes.Compute(int(state.ComputeNodes.Value))
+	}
+	if !state.ComputeMachineType.Unknown && !state.ComputeMachineType.Null {
+		nodes.ComputeMachineType(
+			cmv1.NewMachineType().ID(state.ComputeMachineType.Value),
+		)
+	}
+
+	if !state.AvailabilityZones.Unknown && !state.AvailabilityZones.Null {
+		azs := make([]string, 0)
+		for _, e := range state.AvailabilityZones.Elems {
+			azs = append(azs, e.(types.String).Value)
+		}
+		nodes.AvailabilityZones(azs...)
+	}
+
+	if !state.AutoScalingEnabled.Unknown && !state.AutoScalingEnabled.Null && state.AutoScalingEnabled.Value {
+		autoscaling := cmv1.NewMachinePoolAutoscaling()
+		if !state.MaxReplicas.Unknown && !state.MaxReplicas.Null {
```

Comment:
Consider using `ShouldPatchInt` helper functions
Relevant also for all terraform values check here

Reply:
Not sure it will be clear to use a function for that ...
also in other places in the code, it's already uses that check